tests/vmi_cloudinit_test.go

```diff
@@ -439,10 +434,13 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 				// Expect that the secret is not present on the vmi itself
 				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.UserData).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.UserDataBase64).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.NetworkData).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.NetworkDataBase64).To(BeEmpty())
+
+				testVolume := getCloudInitNoCloudVolume(vmi.Spec.Volumes)
+				Expect(testVolume).ToNot(BeNil(), "should find cloud-init volume in vmi spec")
+				Expect(testVolume.CloudInitNoCloud.UserData).To(BeEmpty())
+				Expect(testVolume.CloudInitNoCloud.UserDataBase64).To(BeEmpty())
+				Expect(testVolume.CloudInitNoCloud.NetworkData).To(BeEmpty())
+				Expect(testVolume.CloudInitNoCloud.NetworkDataBase64).To(BeEmpty())
```

Comment:
I think it will make more sense to check equality with the whole `CloudInitNoCloud` structure. Is it possible?
`Expect(testVolume.CloudInitNoCloud).To(Equal(...))`

Reply:
TBH I am not sure why this assert is necessary at all, this PR doesn't change it.
I prefer to do that on a follow up as other tests follow the same pattern.
It also been said in other thread we should revisit cloud-init tests do some refactoring, including the assertions as you suggest, and remove/consolidate tests.